tests/Tests/UnitTests/Wallets/WalletDataAttribute.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Concordium.Sdk.Tests.UnitTests.Wallets;
+
+/// <summary>
+/// <see cref="EmbeddedResourceDataAttribute"/> used to specify data captured from
+/// wallet key export formats embedded in the manifest as well as a key index, credential
+/// index and a key value. This is used for the theory in <see cref="Xunit"/> tests of
+/// wallets.
+/// </summary>
+public class WalletDataAttribute : EmbeddedResourceDataAttribute
+{
+    private readonly byte _keyIndex;
+    private readonly byte _credentialIndex;
+    private readonly string _key;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="WalletDataAttribute"/> class.
+    /// </summary>
+    /// <param name="walletFilePath">A path to the manifest resource representing the exported wallet keys.</param>
+    /// <param name="credentialIndex">A crenential index.</param>
+    /// <param name="keyIndex">A key index.</param>
+    /// <param name="expectedKey">A string representing the expected key with the corresponding indices.</param>
+    public WalletDataAttribute(
+        string walletFilePath,
```

Comment:
IMHO this attribute is unnecessary complexity.

It does the same as just using `Theory` and `File.ReadXX`. I would just use this.

Reply:
Yea, I responded in another comment.